pandas/core/dtypes/missing.py

```diff
@@ -394,3 +394,10 @@ def na_value_for_dtype(dtype):
     elif is_bool_dtype(dtype):
         return False
     return np.nan
+
+
+def remove_na_arraylike(series):
+    """
+    Return series containing only true/non-NaN values, possibly empty.
+    """
```

Comment:
Let's add tests as well for this function just to make sure it behaves like we want it too.